configure.ac

```diff
@@ -20,19 +22,39 @@ AC_HEADER_STDC
 # Check endianness
 AC_MSG_CHECKING([system endianness])
 AC_PREPROC_IFELSE(
-  [AC_LANG_PROGRAM([[#include <endian.h>]],
-		   [[#if __BYTE_ORDER != __LITTLE_ENDIAN]]
-		   [[#error]]
-		   [[#endif]])],
+  [AC_LANG_PROGRAM([[#if !_AIX]]
+            [[#include <endian.h>]]
+            [[#endif]],
+            [[#if !_AIX]]
+		    [[#if __BYTE_ORDER != __LITTLE_ENDIAN]]
+		    [[#error]]
+		    [[#endif]]
+            [[#endif]])],
   endian=little
   little_endian=1
   big_endian=0)
 
 AC_PREPROC_IFELSE(
-  [AC_LANG_PROGRAM([[#include <endian.h>]],
-		   [[#if __BYTE_ORDER != __BIG_ENDIAN]]
-		   [[#error]]
-		   [[#endif]])],
+  [AC_LANG_PROGRAM([[#if !_AIX]]
+            [[#include <endian.h>]]
+            [[#endif]],
+            [[#if !_AIX]]
+		    [[#if __BYTE_ORDER != __BIG_ENDIAN]]
+		    [[#error]]
+		    [[#endif]]
+            [[#endif]])],
+  endian=big
+  little_endian=0
+  big_endian=1)
+
+AC_PREPROC_IFELSE(
+  [AC_LANG_PROGRAM([[]],
+                [[#if _AIX]]
+		    [[#if __BYTE_ORDER != __BIG_ENDIAN]]
+		    [[#error]]
+		    [[#endif]]
+                [[return 1;]]
+                [[#endif]])],
```

Comment:
This (last block) actually breaks the logic on Linux.  Because this test is always a no-op on Linux, it will never produce an error and so goes on to set endian=big.  So this caused it to report big endian on my little endian system.  It's not a test that causes configure to error out, so it does continue to build.  But we don't want it reporting wrong, of course.